### PR TITLE
docs(operators): add documentation for operator map

### LINKED ISSUE #117: Add docs for operator : transformation - map
Add docs for operator map under group transformation 
 
Below is minimum details needed 
 
- [ ] name 
- [ ] operatorType 
- [ ] signature 
- [ ] parameters 
- [ ] short description 
- [ ] Technical description 
- [ ] Visual example  (Marble diagram) 
- [ ] Real world example (Non-framework) 
- [ ] Related Operators

### PR DESCRIPTION
**Pull request by natmegs**
Add initial version of documentation for creation operator 'empty'. WIP since JSbin is down and cannot test embedded link.
closes #117

### CODE REVIEW

**Comment by ladyleet** [comment]
@natmeg 👍

File: src/operator-docs/transformation/map.ts [hunk 1]
```diff
@@ -1,6 +1,56 @@
...
+    {
+      name: 'Map every click to the clientX position of that click',
+      code: `
+        let clicks = Rx.Observable.fromEvent(document, 'click');
+        let positions = clicks.map(ev => ev.clientX);
+        positions.subscribe(x => console.log(x));
+      `,
+      externalLink: {
+        platform: 'JSBin',
+        url: 'http://jsbin.com/geyokey/embed?js,console'
```
Comment (sumitarora) [inline src/operator-docs/transformation/map.ts]: @natmegs I am seeing this error in JS Bin. Please import library in the bin and update link.
```
"ReferenceError: Rx is not defined
    at geyokey.js:1:19
    at https://static.jsbin.com/js/prod/runner-4.1.1.min.js:1:13850
    at https://static.jsbin.com/js/prod/runner-4.1.1.min.js:1:10792"
```

Comment (ladyleet) [reply src/operator-docs/transformation/map.ts]: thx for helping review this @sumitarora (@natmeg super looking forward to your first PR in this repo!)

Comment (natmegs) [reply src/operator-docs/transformation/map.ts]: Thanks for the feedback! I will fix the bin this evening :)

**Comment by ladyleet** [comment]
thx for helping review @sumitarora! @natmegs let us know how we can help you get to the finish line!

**Comment by natmegs** [comment]
@ladyleet I pushed a change for the JSbin last night! Is there anything left that I've forgotten? Thanks!

**Comment by ladyleet** [comment]
@natmegs can you go ahead and rebase? @sumitarora anything else?

**Comment by natmegs** [comment]
@ladyleet @sumitarora  Alright, I believe the branch is now up to date. If any other changes are needed I'll get them fixed up asap. Thanks for all the help with this one. The next one will be much smoother now!

File: src/operator-docs/transformation/map.ts [hunk 2]
```diff
@@ -1,6 +1,56 @@
 import { OperatorDoc } from '../operator.model';
 
 export const map: OperatorDoc = {
-  'name': 'map',
-  'operatorType': 'transformation'
+  name: 'map',
+  operatorType: 'transformation',
+  signature: 'public map(project: Function, thisArg: any): Observable',
```
Comment (feloy) [inline src/operator-docs/transformation/map.ts]: You could mark optional parameters with question mark:

public map(project: Function, thisArg?: any)

Comment (sumitarora) [reply src/operator-docs/transformation/map.ts]: There is an issue to discuss this https://github.com/ReactiveX/rxjs-docs/issues/168 so this PR should be good to go.

**Comment by ladyleet** [comment]
Woot thanks so much @natmegs for your contribution! :)

**Pull request merged**